appcelerator/titanium_mobile
# fix(android): correct ScrollView.contentOffset units

- Correctly handle `ScrollView.contentOffset` units

##### TEST CASE
- Scroll part way into _green_ and click _red_, this should scroll back to the top
- Scroll part way into _blue_ and click _green_, this should scroll to _green_
- Clicking _blue_ should scroll `100dp` into _red_
```JS
const win = Ti.UI.createWindow();
const scrollView = Ti.UI.createScrollView({
    layout: 'vertical',
    backgroundColor: 'white',
    width: Ti.UI.FILL,
    height: Ti.UI.FILL
});
const view_a = Ti.UI.createView({
    backgroundColor: 'red',
    width: '100%',
    height: '100%'
});
const view_b = Ti.UI.createView({
    backgroundColor: 'green',
    width: '100%',
    height: '100%'
});
const view_c = Ti.UI.createView({
    backgroundColor: 'blue',
    width: '100%',
    height: '100%'
});

view_a.addEventListener('click', e => {
    console.log(`previous contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
    scrollView.contentOffset = {
        x: 0,
        y: 0
    };
    console.log(`new contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
});

view_b.addEventListener('click', e => {
    console.log(`previous contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
    scrollView.contentOffset = {
        x: 0,
        y: view_a.rect.height
    };
    console.log(`new contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
});

view_c.addEventListener('click', e => {
    console.log(`previous contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
    scrollView.contentOffset = {
        x: 0,
        y: '100dp'
    };
    console.log(`new contentOffset: ${JSON.stringify(scrollView.contentOffset)}`);
});

scrollView.add([
    view_a,
    view_b,
    view_c
]);

win.add(scrollView);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-28397)
https://github.com/appcelerator/titanium_mobile/issues/12608
